File: 13 - Roman to integer/solution.kt
class Solution {
    fun romanToInt(s: String): Int {
        var count = 0
        var prev = ' '
        for (i in s.length - 1 downTo 0) {
            count += when (s[i]) {
                'I' -> if (prev == 'V' || prev == 'X') -1 else 1
                'X' -> if (prev == 'L' || prev == 'C') -10 else 10
                'C' -> if (prev == 'D' || prev == 'M') -100 else 100
                'V' -> 5
                'L' -> 50
                'D' -> 500
                'M' -> 1000
                else -> { 0 }
            }
            prev = s[i]
        }
        return count
    }
}

File: 13 - Roman to integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        }
        n = len(s)
        count = roman[s[n - 1]]
        for i in range(n - 1, 0, -1):
            prev = roman[s[i - 1]]
            idx = roman[s[i]]
            if prev >= idx: count += prev
            else: count -= prev
        return count
